oauth2-core/src/main/java/org/springframework/security/oauth2/core/protocol/ResponseAttributesExtractor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.core.protocol;
```

Comment:
I don't think this belongs in the core protocol package. There will be a reactive implementation of the OAuth support that does not support HttpServletRequest. Rather I think we should create a new package that contains converters used for the web